### PR TITLE
preserve normal/dimmed/hidden state depending on youtube section (subscriptions, watch, channel/user or else)

### PR DESCRIPTION
**Pull request by actionless**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -16,8 +17,11 @@
 // You can open new issues at:
 // https://github.com/EvHaus/youtube-hide-watched/issues
 
+/* jshint esversion: 6 */
```
Comment (EvHaus) [inline main.js]: Please remove this. There's no need to use JSHint when ESLint is already used.

File: package.json [hunk 2]
```diff
@@ -14,13 +14,15 @@
...
+		"jshint": "jshint main.js",
+		"lint": "yarn packagelint && yarn eslint && yarn jshint",
 		"packagelint": "prettier-package-json --use-tabs --write ./package.json",
 		"test": "yarn lint"
 	},
 	"devDependencies": {
 		"eslint-config-globex": "^2.7.1",
-		"prettier-package-json": "^2.1.3"
+		"prettier-package-json": "^2.1.3",
+		"jshint": "^2.11.0-rc1"
```
Comment (EvHaus) [inline package.json]: Please remove this. There's no need to use JSHint when ESLint is already used.

**Comment by actionless** [comment]
i'm not familiar with javascript, but in python it's usually better to use both flake8 and pylint, as sometimes one could catch some mistake which other wouldn't

regarding functionality -- should i make some way to choose, if normal/dimmed/hidden state would be applied globally or only to the current section?

File: main.js [hunk 3]
```diff
@@ -176,12 +194,15 @@ html[dark] .YT-HWV-BUTTON {
 		// [TODO] If we're on the Trending page -- we don't support it yet.
```
Comment (actionless) [inline main.js]: also i could imlpement it for "trending" section while i'm already on this

Comment (EvHaus) [reply main.js]: Sure. That would be great.

**Comment by EvHaus** [comment]
> i'm not familiar with javascript, but in python it's usually better to use both flake8 and pylint, as sometimes one could catch some mistake which other wouldn't

In Javascript it's a bit different. JSHint is very very old. It's not really used in the community anymore. ESLint can do everything that JSHint can do, and more. We don't need to use JSHint at all.

> regarding functionality -- should i make some way to choose, if normal/dimmed/hidden state would be applied globally or only to the current section?

That would be ideal, but I'm not sure where we would put such UI to handle the settings.

**Comment by actionless** [comment]
thanks for the info regarding jshint, i'll update to remove it

i neither have any good ideas regarding the UI, 2 which come up in mind:
1) add a dropdown arrow right to the "eye" icon of this extension
2) inject "Hide-Watched Preferences" into this menu:
2020-01-03--1578077701_556x1374_scrot

**Comment by actionless** [comment]
3) refactor it into WebExtension, and use settings page of the extension

**Comment by EvHaus** [comment]
Those are the 3 ideas I had as well. I think #1 is probably the easiest.

Let's just keep the code as you have it now. We can work on #1 as a new feature later.

Let me know when you've made your changes and I can work on getting this released.

**Comment by actionless** [comment]
i've updated

**Pull request merged**